tests/test_sessions.py

```diff
@@ -53,7 +53,7 @@ def test_generate_key(self):
             key = session.generate_key(pkcs11.KeyType.AES, 128,
                                        label='MY KEY',
                                        id=b'\1\2\3\4',
-                                       capabilities=0)
+                                       template={pkcs11.Attribute.ENCRYPT: False})
```

Comment:
Hmm.. does this break an assumption of the library, if you're not set the capability is that false or default?

Maybe we're thinking about this all wrong, and you get the default OR the capabilities you've set only, without merging.

So `capabilities=None` -> default set of True's, no falses. `capabilities is not None` Trues for set capabilities. No falses. That way this test won't have to be changed. And for anything not specified you get the HSM's default, which should be sensible.

Reply:
For this test we don't know the HSM defaults so we explicitly need to set ENCRYPT to False, this is achieved by the current behaviour which set all flags to False.

Not sure what you mean by "So capabilities=None -> default set of True's, no falses. capabilities is not None Trues for set capabilities. No falses." Won't we still have to force the flag to False somehow?

Reply:
I'm not sure what I mean either. I think, I've been thinking about this all wrong.

We've got 3 states: _true_, _false_ and _left-out_ (default). In this test we wanted to pass `False` so we unset the capability. That is intuitive to me.

This has worked for everything except for HSMs that don't accept the Attribute at all. There have been a few times where it could have been useful to remove a value out of the template. Perhaps what we need is a new constant `DEFAULT`, which is filtered out of the final merged template.

Something like this:

```python
template={
    pkcs11.Attribute.UNWANTED_ATTRIBUTE: pkcs11.DEFAULT
}
```

We could then filter out all `DEFAULT` values from the remaining template. I'm happy to code this up if you're not keen.

Reply:
So I have tried to code this up - there is no change to the tests, `capabilities=N` does what it does before. If the user really does not want the attr sent to the HSM, the library can be overridden with the semantics you suggest:

```
# my HSM does not want ENCRYPT VERIFY WRAP on private keys
obj.unwrap_key( .... template={Attribute.ENCRYPT: pkcs11.DEFAULT,
    Attribute.VERIFY: pkcs11.DEFAULT,
    Attribute.WRAP: pkcs11.DEFAULT})
```